mocca-http-client-tests/src/main/java/com/paypal/mocca/client/BasicMoccaHttpClientTest.java

```diff
@@ -86,12 +78,45 @@ void testReadTimeout() {
                 assertEquals(e.getCause().getCause().getClass(), expectedTimeoutExceptionCause());
             }
         }
+
+        protected static final int testClientInstanceReadTimeout_CLIENT_READ_TIMEOUT = 5000;
+
+        /**
+         * This will perform a test where the GraphQL call respects client instance specified HTTP read timeout (i.e. per request timeout).
```

Comment:
Don't you mean that the the `MoccaClient` timeouts are respected?  I read this sentence to mean that the underlying client timeouts will be used, which the next sentence contradicts.

RE the test itself..

Minimally, I think it should share the code with `testReadTimeout`.  I also think a little better way to do this would be to have an abstract `create` overload like:

```
/**
* Use the given timeouts to configure the underlying HTTP client library used to create the MoccaHttpClient.  The goal is to show that MoccaClient builder stuff overrides these values
* in the client library.
*/
abstract MoccaHttpClient.WithRequestTimeouts create(Duration socketTimeout, Duration connectTimeout);
```

and then pair that with a base class test.  The class extensions would just need to implement the create(Duration, Duration).  Basically the same pattern we use for `testReadTimeout`.  That + javadoc on the create should make it pretty clear on what class extensions need to do to verify performance.  It's a little confusing to have 2 different patterns that basically solve the same problem.. 

In neither case do I think you need to make the test generic.  In the code as you have it, you don't use the added abstract method, do you?

This isn't some huge deal though.  If this is how you want it, I won't let this block progress.

Reply:
Done. Indeed I didn't use the overloaded create method. I tried a few different things before I got to that point and I didn't realize was in the end not using it. Fixed now.

Reply:
I also fixed the method comment and consolidated those two test methods into one.